test/request.js

```diff
@@ -2294,26 +2337,17 @@ describe('Request', () => {
                 method: 'POST'
             };
 
-            await new Promise(async (resolve) => {
-
-                const req = Http.request(options, (res) => {
-
-                    expect([503, 408]).to.contain(res.statusCode);
-                    expect(timer.elapsed()).to.be.at.least(80);
-                    resolve();
-                });
+            const res = await new Promise((resolve, reject) => {
```

Comment:
This seems wrong. `req` errors will effectively be ignored since they would trigger after the `response` event, rejecting into an already resolved `Promise`.

Reply:
That is true.  But we do know that the test will already be over, so it would be a late error.  This was occurring when stopping the server causes the client to experience a remotely closed connection (presumably racing with `req.end()`?), and sometimes it would land while a future test was running.  I will try to handle it a bit better.